modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[APIRest]APIGenerator/ExposeQueryParametersInAPIExplorerForGeneratedAPI.testcase

```diff
@@ -0,0 +1,548 @@
+
+			PropertyAPI.createAPIProperty(
+				name = "entryName",
+				objectFieldERC = "studentName",
+				schemaId = ${schemaId});
+		}
+
+		task ("And Given I create endpoint a sort with name:asc") {
+			var endpointId = JSONUtil.getWithJSONPath(${response}, "$.apiApplicationToAPIEndpoints[*].id");
+
+			SortAPI.createAPISort(
+				endpointId = ${endpointId},
+				sortValue = "name:asc");
+		}
+
+		task ("When execute get{Endpoint}Page with sort {property}:desc in API Explorer") {
+			APIExplorer.navigateToOpenAPI(customObjectPlural = "my-app");
+
+			APIExplorer.executeAPIMethod(
+				method = "getTestEndpointPage",
+				parameter = "sort",
+				service = "testSchema",
+				value = "entryName:desc");
+		}
+
+		task ("Then I can see items in desc sort order in response") {
+			AssertTextEquals.assertPartialText(
+				locator1 = "OpenAPI#RESPONSE_BODY",
+				method = "getTestEndpointPage",
+				value1 = "\"items\": [     {       \"entryName\": \"Tom\"     },     {       \"entryName\": \"Test\"     },     {       \"entryName\": \"Able\"     }   ]");
+		}
+	}
+
+	@priority = 4
+	test CannotSeeFilterAndSortParametersForSingleElementEndpoint {
+		property portal.acceptance = "true";
+
+		task ("Given an published API application with singleElement endpoint") {
+			ApplicationAPI.createAPIApplication(
+				baseURL = "my-app",
+				endpointName = "testEndpoint",
+				endpointPath = "/testEndpoint/{id}",
+				pathParameter = "ID",
+				relatedEndpoint = "true",
+				retrieveType = "singleElement",
+				status = "published",
+				title = "My-app");
+		}
+
+		task ("When I expand get{EndpointPath} in /o/c/{aPIApplication}") {
+			APIExplorer.navigateToOpenAPI(customObjectPlural = "my-app");
+
+			APIExplorer.expandAPIMethod(
+				method = "getTestEndpointId",
+				service = "default");
+		}
+
+		task ("Then filter and sort parameters are not present") {
+			AssertElementNotPresent(
+				locator1 = "OpenAPI#PARAMETER_UNDER_METHOD",
+				method = "getTestEndpointId",
+				parameter = "filter");
+
+			AssertElementNotPresent(
+				locator1 = "OpenAPI#PARAMETER_UNDER_METHOD",
+				method = "getTestEndpointId",
+				parameter = "sort");
+		}
+	}
+
+	@priority = 4
+	test CanSeeFilterAndSortParametersInComapnyScopedEndpoint {
+		property portal.acceptance = "true";
+
+		task ("Given an published API application with company scoped endpoint") {
+			ApplicationAPI.createAPIApplication(
+				baseURL = "my-app",
+				endpointName = "testEndpoint",
+				endpointPath = "/testEndpoint",
+				relatedEndpoint = "true",
+				status = "published",
+				title = "My-app");
+		}
+
+		task ("When I expand get{endpoint}Page in /o/c/{aPIApplication}") {
+			APIExplorer.navigateToOpenAPI(customObjectPlural = "my-app");
+
+			APIExplorer.expandAPIMethod(
+				method = "getTestEndpointPage",
+				service = "default");
+		}
+
+		task ("Then I can see filter and sort parameters with string type are present") {
+			AssertElementPresent(
```

Comment:
I think it doesn't assert the 'filter' and 'sort' are in 'String' type. Could you please check and update? Thanks.